Nf test/gbcms

buehlere opened the pull request:
## PR checklist

Addresses one of the items in #15 and #9 

- [x] adding GBCMS module and config files for nf-test
- [x] If you've fixed a bug or added code that should be tested, add tests!
- [x] If you've added a new tool - have you followed the module conventions in the [contribution docs](https://github.com/nf-core/modules/tree/master/.github/CONTRIBUTING.md)
- [x] If necessary, include test data in your PR.
- [x] Remove all TODO statements.
- [x] Emit the `versions.yml` file.
- [x] Follow the naming conventions.
- [x] Follow the parameters requirements.
- [x] Follow the input/output options guidelines.
- [x] Add a resource `label`
- [x] Use BioConda and BioContainers if possible to fulfil software requirements.
- Ensure that the test works with either Docker / Singularity. Conda CI tests can be quite flaky:
  - [x] nf-test test --profile=docker --tag <MODULE> 
  - [x] nf-test test --profile=singularity --tag <MODULE> 
  - [ ] nf-test test --profile=conda --tag <MODULE>  (excluded from nf-test, not planning to support at this time) 


We'll need to use this to test:
`nf-core modules --git-remote https://github.com/mskcc-omics-workflows/modules.git -b feature/gbcms test gbcms`

ErikDanielsson committed:
Add fastp and remove fastqc

ErikDanielsson committed:
Remove input sections

mirpedrol committed:
restructure modules folder

mirpedrol committed:
mimic old sha eb4bc244de7eaef8e8ff0d451e4ca2e4b2c29821

mirpedrol committed:
mimic las commit

mashehu committed:
Merge branch 'branch-tester-restructure' into 'branch-tester'

Branch tester restructure

See merge request nf-core/modules-test!2

mashehu committed:
Update .nf-core.yml

mirpedrol committed:
add a new org_path other than nf-core

mirpedrol committed:
add test files

mirpedrol committed:
add pytest_modules.yml

mirpedrol committed:
fix path

mirpedrol committed:
fix fastp name :(

mirpedrol committed:
fix relative paths

mirpedrol committed:
add nextflow.config

mirpedrol committed:
add nextflow.config

mirpedrol committed:
add bam_stats_samtools

mirpedrol committed:
add samtools

Commit:
init

Commit:
working on stub test

Commit:
adjusting stub test

Commit:
adding basic testing

buehlere committed:
Merge branch 'develop' into feature/gbcms

Commit:
Update nf-test.yml

Commit:
Update main.nf

Commit:
Update main.nf

Commit:
using wrong test.yml, removing and correcting

Commit:
Merge remote-tracking branch 'origin/develop' into feature/gbcms

Commit:
delete old test yml

Commit:
prettier

Commit:
updating module inputs and meta.yml

Commit:
fixing up test

Commit:
Update meta.yml

Commit:
fix linting, docs and outputs

Commit:
resolve merge

Commit:
fixing docs

Commit:
Update meta.yml

Commit:
more test refactoring

Commit:
Merge branch 'feature/gbcms' of https://github.com/mskcc-omics-workflows/modules into feature/gbcms

Commit:
fixing up tests and inputs from code review

Commit:
fix test output

Commit:
fix outputs

Commit:
fix outputs retry

Commit:
get rid of old params

Commit:
dropping params in nextflow.config

Commit:
fixing verion name

Commit:
fixing verion name retry

Commit:
removing conda test for gbcms

Commit:
removing conda test for gbcms

Commit:
removing conda test for gbcms

Commit:
adjusting container variable and undoing changes to test.yml

Commit:
adjusting script to deal with conda case

Commit:
gbcms

Commit:
fixing exclude tags

Commit:
Update test.yml

Commit:
remove old argument, config spacing

Commit:
Update test.yml

Commit:
ignore DS_Store

Commit:
basic nf-test for gbcms

Commit:
changing tag

Commit:
upgrading test

Commit:
re-formatting

Commit:
master -> main

Commit:
foramatting and remove old tests

Commit:
remove gbcms from pytest.yml

Commit:
updates for linting

Commit:
adjusting for conda test

Commit:
ignoring conda test

Commit:
removing conda

anoronh4 commented on modules/msk/gbcms/main.nf:
```diff
@@ -0,0 +1,58 @@
+    def prefix = task.ext.prefix ?: "${meta.id}"
+    """
+    echo stub > 'variant_file.maf'
+    echo "${task.process}:" > versions.yml
```
why not use the same syntax as the non-stub block? looks like the second line also won't be indented, which might not be a problem for stub tests, but might be better to follow the same formatting just in case.

buehlere replied:
agreed. I just updated the syntax

anoronh4 commented on modules/msk/gbcms/tests/main.nf.test:
```diff
@@ -0,0 +1,45 @@
+
+        // (the module requires running more than one process to generate the required output)
+        // add the 'setup' method here.
+        // You can find more information about how to use a 'setup' method in the docs (https://nf-co.re/docs/contributing/modules#steps-for-creating-nf-test-for-chained-modules).
```
seems like part of the TODO block, should we remove?

buehlere replied:
I removed these

anoronh4 reviewed (commented):
i left some comments, i also see a folder gbcms_old which was probably included unintentionally.

Commit:
updating stub and removing old comments

:heavy_check_mark: cleaning up stub test and old comments

Commit:
Update main.nf

Commit:
updating test.yml

we want the tests to run on PRs to main and develop

buehlere commented:
> i left some comments, i also see a folder gbcms_old which was probably included unintentionally.

I removed those files. I also realized I needed to update the test.yml so we test on PRs to main and develop: https://github.com/mskcc-omics-workflows/modules/blob/9d49dca37efe01e0a9411a768e3e3f3e89ed08d8/.github/workflows/test.yml#L5-L6. Let me know how things look!

buehlere committed:
Update test.yml

✔️ adding back conda test

anoronh4 commented on .github/workflows/test.yml:
```diff
@@ -217,9 +450,186 @@ jobs:
+          - profile: "conda"
+            tags: vt/decompose
+          - profile: "conda"
+            tags: subworkflows/vcf_annotate_ensemblvep
```
do you plan to update the conda test? if not we should we should probably exclude the CI test by adding it to this list of exclusions. that way all tests pass. i think we can just add something like 
```
- profile: "conda"
  tags: gbcms
```

buehlere replied:
No plans as of right now. I added conda gbcms to the exclude block. 

I also removed the nf-core modules from the exclude block.

anoronh4 commented on tests/config/nf-test.config:
```diff
@@ -0,0 +1,50 @@
+    }
+    docker {
+        docker.enabled = true
+        docker.userEmulation = false
```
nf-core removed the docker.userEmulation option recently https://github.com/nf-core/modules/pull/4626

anoronh4 commented on tests/config/nf-test.config:
```diff
@@ -0,0 +1,50 @@
+    }
+    podman {
+        podman.enabled = true
+        podman.userEmulation = true
```
nf-core removed the `podman.userEmulation` option recently https://github.com/nf-core/modules/pull/4626

buehlere replied:
good catch. I updated `tests/config/nf-test.conf` and t`ests/config/nextflow.config` to remove the <docker|podman>.userEmulation argument.

Commit:
removing conda test, updating config

:heavy_check_mark: from review on #10, updating nextflow config and ignoring conda test on module

Commit:
remove user emulation

Commit:
remove nf-core modules

buehlere commented:
@anoronh4 I think this is ready for another look and maybe ready to merge? Let me know your thoughts!

nikhil commented:
I have a PR for SNP pileup ready, I am going to wait on this one for the testing framework.

anoronh4 commented:
@buehlere looks good from my end but i don't see any "approve this" button. feel free to re-request my review, and i'll rubber-stamp it, or just go ahead and merge, either way looks good to me.

buehlere commented:
> @buehlere looks good from my end but i don't see any "approve this" button. feel free to re-request my review, and i'll rubber-stamp it, or just go ahead and merge, either way looks good to me.

sounds good and thanks! Just merged to develop. 

@nikhil you should be good to merge this into your module.
